L1Trigger/TrackFindingTMTT/interface/HTcell.h

```diff
@@ -0,0 +1,163 @@
+#ifndef L1Trigger_TrackFindingTMTT_HTcell_h
+#define L1Trigger_TrackFindingTMTT_HTcell_h
+
+#include "L1Trigger/TrackFindingTMTT/interface/Utility.h"
+#include "L1Trigger/TrackFindingTMTT/interface/Stub.h"
+#include "FWCore/Utilities/interface/Exception.h"
+
+#include <vector>
+#include <map>
+#include <algorithm>
+#include <utility>
+
+using namespace std;
+
+//=== A single cell in a Hough Transform array.
+
+namespace tmtt {
+
+  class Settings;
+  class Stub;
+
+  class HTcell {
+  public:
+    HTcell() {}
+    ~HTcell() {}
+
+    // Initialization with cfg params,
+    // sector number, rapidity range of current sector, estimated q/Pt of cell,
+    // and the bin number of the cell along the q/Pt axis of the r-phi HT array,
+    // and a flag indicating if this cell is the merge of smaller HT cells.
+    void init(const Settings* settings,
+              unsigned int iPhiSec,
+              unsigned int iEtaReg,
+              float etaMinSector,
+              float etaMaxSector,
+              float qOverPt,
+              unsigned int ibin_qOverPt,
+              bool mergedCell,
+              bool miniHTcell = false);
+
+    // Add stub to this cell in HT array.
+    void store(const Stub* stub) { vStubs_.push_back(stub); }
+
+    // Add stub to this cell in HT array, indicating also which subsectors within the sector is consistent with.
+    void store(const Stub* stub, const vector<bool>& inSubSecs) {
+      this->store(stub);
+      subSectors_[stub] = inSubSecs;
+      if (inSubSecs.size() != numSubSecs_)
+        throw cms::Exception("HTcell: Wrong number of subsectors!");
+    }
+
+    // Termination. Search for track in this HT cell etc.
+    void end();
+
+    //=== Cfg of cell
+
+    // Does this HTcell correspond to merged HT cells (e.g. 2x2)?
+    bool mergedCell() const { return mergedCell_; }
+
+    //=== Get results
+    //=== Most of these functions operate on the filtered stubs, which are stubs passing any requested stub filters (e.g. bend filter).
+    //=== If no filters were requested, they are identical to the unfiltered stubs.)
+
+    // Get filtered stubs in this cell in HT array.
+    const vector<const Stub*>& stubs() const { return vFilteredStubs_; }
+
+    // Check if a specific stub is in this cell and survived filtering.
+    bool stubInCell(const Stub* stub) const {
+      return (std::count(vFilteredStubs_.begin(), vFilteredStubs_.end(), stub) > 0);
```

Comment:
As the vector only has about 6 entries, this doesn't give significant CPU saving. And if we change from vector to unordered_set here, we would have to make the same change at many other points in the code to be consistent. If OK with you, this doesn't seem worth while?

Reply:
`std::find()` would still give better average performance than `std::count()` with an unsorted vector

Reply:
For "about 6 entries" an unsorted vector with `std::find()` is indeed likely the best way to go.

Is there a well-defined upper bound for the number of entries, or can the vector in principle grow without bounds? I just thinking that with about 6 (pointer) entries the overhead of `vector` is about 33 % (because the `vector` itself uses 3 pointers). If there is a well-defined upper bound that is not "much larger" than 6 there are alternatives for `vector`, but if it can grow without bounds, it would be difficult to avoid `vector`. On the other hand this may be micro-optimization that can be done later as well.

Reply:
actually for small collection sizes (up to 15~16 or so) a `vector` is faster than an `unordered_set`:
![image](https://user-images.githubusercontent.com/4069793/80823708-66fb4d00-8bdd-11ea-9c02-433a6cc968d2.png)

Reply:
This is interesting, but it remains the case that the CPU use here is negligible, and that if we change this container from vector to something else, we would have to propagate this change to many other parts of the code. If OK with you, I therefore propose to leave this unchanged.

Reply:
I think the conclusion (especially thanks to @fwyzard's test) was that for "about 6 entries" (non-sorted) vector is preferred anyway.